### PR TITLE
add event

### PR DESCRIPTION
**Pull request by davis999**
add design doc for event.

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#### 4.1.2 Topic Design
+
+* Topic name: `reactivesw-order-created`
+* Subscription name: `inventory-order-created`
+
+#### 4.1.3 Workflow
+
+1. get list of InventoryRequest from event message
+2. validate orderId: if orderId is exist, stop this process
```
Comment (kaka89) [inline docs/design.md]: acknowledge

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* Topic name: `reactivesw-order-created`
+* Subscription name: `inventory-order-created`
+
+#### 4.1.3 Workflow
+
+1. get list of InventoryRequest from event message
+2. validate orderId: if orderId is exist, acknowledge to blocker, stop this process
+3. remove inventory: remove sku inventory in one transaction, acknowledge to blocker
+4. store orderId
+5. send event
```
Comment (kaka89) [inline docs/design.md]: store event

File: docs/design.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+| field name | field type | comment |
+|---|---|---|
+| orderId | String | |
+| result | Boolean  | true when remove inventory successful |
+| message | String | |
+
+#### 4.2.2 Topic
+
+Topic name: `reactivesw-inventory-removed`
```
Comment (kaka89) [inline docs/design.md]: work flow

**Review by kaka89 — APPROVED** [review]

File: src/main/java/io/reactivesw/inventory/application/model/OrderCreationEvent.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import lombok.Setter;
+
+import java.util.List;
+
+/**
+ * The type Order creation event.
+ */
+@Getter
+@Setter
+public class OrderCreationEvent {
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/application/model/OrderCreationEvent.java:15]: OrderCreatedEvent

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * The type Order creation consumer.
+ */
+@Service
+public class OrderCreationConsumer {
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java:22]: OrderCreatedConsumer

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  /**
+   * log.
+   */
+  private static final Logger LOG = LoggerFactory.getLogger(OrderCreationConsumer.class);
+
+  /**
+   * message consumer.
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java]: Message

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationEventHandler.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    this.reservedOrderService = reservedOrderService;
+    this.eventMessageService = eventMessageService;
+  }
+
+  /**
+   * Handle order creation.
+   *
+   * @param event the event
+   */
+  public void handleOrderCreation(OrderCreationEvent event) {
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/application/service/OrderCreationEventHandler.java]: transaction?

File: src/main/java/io/reactivesw/inventory/infrastructure/enums/EventStatus.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
+package io.reactivesw.inventory.infrastructure.enums;
+
+/**
+ * Event status.
+ */
+public enum EventStatus {
+  /**
+   * new created events.
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/infrastructure/enums/EventStatus.java]: UP case

File: src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+/**
+ * Created by Davis on 17/4/20.
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java]: Header

File: src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+public final class ReservedOrderSpecification {
+
+  /**
+   * Instantiates a new Reserved order specification.
+   */
+  private ReservedOrderSpecification() {
+  }
+
+  /**
+   * specification for fetch Events.
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java]: UP case

File: src/main/java/io/reactivesw/inventory/infrastructure/util/EventTopics.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * Event Topics.
+ */
+public final class EventTopics {
+
+  /**
+   * The INVENTORY_REMOVED topic.
+   */
+  public static final String INVENTORY_REMOVED = "reactivesw-inventory-removed";
```
Comment (kaka89) [inline src/main/java/io/reactivesw/inventory/infrastructure/util/EventTopics.java]: reserved

File: docs/design.md (hunk 3 above)
Comment (davis999) [reply docs/design.md]: done.

File: src/main/java/io/reactivesw/inventory/application/model/OrderCreationEvent.java (hunk 4 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/application/model/OrderCreationEvent.java:15]: OrderCreationEvent is better.

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java (hunk 5 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java:22]: Creation is better.

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java (hunk 6 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/application/service/OrderCreationConsumer.java]: done.

File: src/main/java/io/reactivesw/inventory/application/service/OrderCreationEventHandler.java (hunk 7 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/application/service/OrderCreationEventHandler.java]: done.

File: src/main/java/io/reactivesw/inventory/infrastructure/enums/EventStatus.java (hunk 8 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/infrastructure/enums/EventStatus.java]: done.

File: src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java (hunk 9 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java]: done.

File: src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java (hunk 10 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/infrastructure/repository/ReservedOrderSpecification.java]: done.

File: src/main/java/io/reactivesw/inventory/infrastructure/util/EventTopics.java (hunk 11 above)
Comment (davis999) [reply src/main/java/io/reactivesw/inventory/infrastructure/util/EventTopics.java]: done.
change it in design doc too.

**Review by kaka89 — APPROVED** [review]

**Pull request merged**